tee-worker/litentry/core/assertion-build/src/nodereal/spaceid/mod.rs

```diff
@@ -0,0 +1,226 @@
+// Copyright 2020-2023 Trust Computing GmbH.
+// This file is part of Litentry.
+//
+// Litentry is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Litentry is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Litentry.  If not, see <https://www.gnu.org/licenses/>.
+
+#[cfg(all(feature = "std", feature = "sgx"))]
+compile_error!("feature \"std\" and feature \"sgx\" cannot be enabled at the same time");
+
+#[cfg(all(not(feature = "std"), feature = "sgx"))]
+extern crate sgx_tstd as std;
+
+pub mod bnb_domain_holding_amount;
+
+use crate::*;
+use lc_data_providers::{
+	nodereal::{DomainInfo, NoderealClient, SpaceIDApiList},
+	DataProviderConfigReader, ReadDataProviderConfig,
+};
+use serde::{Deserialize, Serialize};
+use std::string::ToString;
+
+pub struct SPACEIDLogic;
+pub trait SPACEIDLogicInterface {
+	fn get_bnb_domain_holding_amount(addresses: &[String]) -> Result<usize>;
+}
+
+impl SPACEIDLogicInterface for SPACEIDLogic {
```

Comment:
Similar here as well, I am thinking we can try to remove SPACEID everywhere, except maybe in some of the assertion product does need us add spaceid (but I did check and didn't see any of that in the VC requirement/assertions etc), then our new VC is simply a domain holding VC, that's completely not limited to SPACEID. Make sense? @zhouhuitian

Reply:
sure, i will re-adjust the structure.